pkg/repositories/interfaces/reservation_repo.go

```diff
@@ -0,0 +1,19 @@
+package interfaces
+
+import (
+	"context"
+	"time"
+
+	"github.com/flyteorg/datacatalog/pkg/repositories/models"
+)
+
+// Interface to interact with Reservation Table
+type ReservationRepo interface {
+
+	// Get reservation
+	Get(ctx context.Context, reservationKey models.ReservationKey) (models.Reservation, error)
+
+	// Create the reservation. If the reservation already exists, we try to take over the
+	// reservation via update.
```

Comment:
Please add to the docs here a more detailed explanation of how that happens... and the role `expires_at` plays as well as the conditional write we do...

Reply:
done